src/tree-manager/flatten-tree.js

```diff
@@ -141,39 +146,58 @@ function walkNodes({ nodes, list = new Map(), parent, depth = 0, simple, showPar
       node._id = node.id || `${rootPrefixId ? `${rootPrefixId}-${i}` : i}`
     }
 
-    if (node.isDefaultValue) {
-      defaultValues.push(node._id)
+    if (single && node.checked) {
```

Comment:
I tried a few different ways to see if we can somehow make this any simpler. With our current "First Person Wins" and "`defaultValue` supersedes `checked`" strategy, things end up being complicated since we need to backtrack.

As we walk the forest/tree, we are moving forward, so things become a lot simple if we switch to "Last Person Wins" strategy. 

Something like: 

```
  walk 
    if isDefaultValue
        set node checked

    if node is checked 
       if single
          checkNodes[0].checked = false
          // optionally throw a warning to the user maybe?? Feel free to change the wording if you like...
          if checkNodes[0].isDefaultValue
             console.warn("You've assigned more than 1 default values while the component is in single select mode (you've set either `simpleSelect=true` or `radioSelect=true`). Only the last node will be honored. If you intend to use multiple values, switch to multi-select instead.")
              // not sure if we should clean up the input here by setting isDefaultValue to false
          
          // update with latest node since Last Person Wins
          checkNodes = [node]
          
       else 
          checkNodes.push(node)
```

It _should_ bring down the Cognitive complexity score as well. I as too exhausted to try this last night so I'm not sure if this will work 100% as expected but worth a try IMHO. Let me know your thoughts.

Reply:
Give it a go. Won't you have problem distinguishing a previous checked node which was done by isDefaultValue then though? If isDefaultValue should no longer supersede checked, or vice versa,  then the current logic would be simplified also.

Reply:
This is just a convenience as most of the issues can be resolved by setting the proper data, so I don't want to go crazy on this. Certainly not worth sacrificing performance and increasing the complexity.

We can simply go with "last person wins" or "first person wins" with whatever takes the simplest of solutions. 

> If isDefaultValue should no longer supersede checked, or vice versa, then the current, "First person wins" logic could be simplified also.

With you. In order to both simplify logic and make sure we're clear about the behavior, what do you say to this?

(only for **single** select cases since this is a non-issue for multi-select )

 1. Setting default values - First `isDefaultValue` wins (helps us shortcircuit the tree iteration)
 2. Setting checked values - Last person wins (we walk all nodes anyway and is easier to override previous values) don't know if we wanna go with a universal theme of FPS or LPS in all cases
 3. When both defaultValues/checked present - treat defaultValue nodes as checked and then follow `2`

Will try this and see where I land. Just thinking out loud here so I'm sure I'll find something while implementing and testing.

Reply:
I tried this and I don't see it being any simpler. I think what we have right now is pretty good. Unless you have any objections, this is good to be merged.